whelktool/scripts/cleanups/2019/04/remove-duplicated/from-list-move-items-and-delete-instance-duplicates.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * This script moves holdings from one instance to another specified by
+ * the file id_lxl-2281_eboksberikning.txt The holdings are moved from
+ * the instance in the right column to the instance in the left column.
+ *
+ * See LXL-2380 for more info.
+ *
+ */
+
+PrintWriter failedBibIDs = getReportWriter("failed-to-delete-bibIDs")
+PrintWriter scheduledForChange = getReportWriter("scheduledForChange")
+PrintWriter instances = getReportWriter("instances")
+
+def file = new File(scriptDir, "id_lxl-2281_eboksberikning.txt");
+
+file.eachLine { line ->
+    def (goodInstanceId, badInstanceId) = line.split(";").collect { it.trim() }
+
+    def fullGoodInstanceId = "http://libris.kb.se/bib/${goodInstanceId}" as String
+
+    //Create a map of sigels mapping to items held by the good instance
+    def sigelToGoodHoldingMap = [:]
+
+    selectByIds([fullGoodInstanceId]) { goodBib ->
+        instances.println("Good bib ${fullGoodInstanceId} is a ${goodBib.graph[1][TYPE]}")
+        selectBySqlWhere("""
+                data#>>'{@graph,1,itemOf,@id}' = '${goodBib.graph[1][ID]}'
+        """, silent: true) { hold ->
+            sigelToGoodHoldingMap.put(hold.graph[1].heldBy[ID], hold)
```

Comment:
There *should* only ever be one holding for each sigel, but it might be nice to log if this assumption is incorrect. Also, if there are more than one holding for a specific sigel, we'll overwrite here which will lead us to not move all holdings (which will cause the deletion of the bad instance to fail).